### PR TITLE
SetupPayload parsers for Android apps

### PR DESCRIPTION
**Pull request by vidhis88**
Adding JNI-bridging code for SetupPayload package and automake rules to build SetupPayloadParser.jar and libSetupPayloadParser.so for use in Android apps.

### CODE REVIEW

**Review by andy31415 — COMMENTED** [review]
Can we avoid including binary libraries like libc++_shared.so and such?

I would expect us to build any required libaries rather than including the binary data.

**Review by gerickson — CHANGES_REQUESTED** [review]
+1 to @andy31415's comments about including .so binaries.

Also, this seems like a good time to merge in openweave-core's `--{disable,enable}-java` options from `configure.ac`.

**Review by woody-apple — APPROVED** [review]

**Review by gerickson — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -12,7 +12,7 @@ Makefile.in
 aclocal.m4
 autom4te.cache
 output/
-/build/
+/build/autoconf/
```
Comment (andy31415) [inline .gitignore:15]: Ignoring the entire build seemed correct to me: we have build/default and such

**Review by andy31415 — APPROVED** [review]

**Comment by woody-apple** [comment]
@randyrossi can you rebase this please (and fix restyle)?

File: .gitignore (hunk 1 above)
Comment (vidhis88) [reply .gitignore:15]: I added a build/make/java.mk file. Hence, modified this to ignore the only other folder inside build/ similar to the openweave organization.
Would you suggest I move this java.mk file to a different spot?

**Comment by vidhis88** [comment]
Closing this PR out in favor of https://github.com/project-chip/connectedhomeip/pull/1454 as this one was from before when the CHIP repo was private. Trying to rebase off of the public repo's master is proving to be more of a hassle.
Hence, created a fresh branch with the same changes from my public fork of the CHIP repo.
Apologies for re-review.

File: .gitignore (hunk 1 above)
Comment (vidhis88) [reply .gitignore:15]: Have moved the java.mk file outside of build folder, thus, voiding the need for this change.
This is done in https://github.com/project-chip/connectedhomeip/pull/1454 from my fork off of the public CHIP repo (this PR is from before, when the repo was private and updating there is turning out to be more of a hassle).

File: src/inet/InetInterface.cpp [hunk 2]
```diff
@@ -51,11 +51,7 @@
 #endif /* HAVE_SYS_SOCKIO_H */
 #include <net/if.h>
 #include <sys/ioctl.h>
-#ifdef __ANDROID__
```
[the remark this answers is not in the document]
Comment (vidhis88) [reply src/inet/InetInterface.cpp:54]: Done as part of https://github.com/project-chip/connectedhomeip/pull/1411

**Pull request closed without merging**